specification/e-referrals-service-api.yaml

```diff
@@ -2343,7 +2343,63 @@ paths:
         '429':
           $ref: 'components/schemas/responses/TooManyRequests.yaml'
         '500':
-          $ref: 'components/schemas/responses/InternalServerError.yaml' 
+          $ref: 'components/schemas/responses/InternalServerError.yaml'
+  /STU3/ReferralRequest/{ubrn}/$ers.cancelReferral:
+    post:
+      security:
+        - bearerAuth: []
+      description: |
+        ## Overview
+         TBD
+      summary: A036 - Cancel referral
+      operationId: a036-cancel-referral
+      parameters:
+        - $ref: 'components/schemas/headers/request/BearerAuthorization.yaml'
+        - $ref: 'components/schemas/headers/request/OdsCode.yaml'
+        - $ref: 'components/schemas/headers/request/BusinessFunction.yaml'
+        - $ref: 'components/schemas/headers/request/CorrelationID.yaml'
+        - $ref: 'components/schemas/path/Ubrn.yaml'
+        - $ref: 'components/schemas/headers/request/IfMatch.yaml'
+      requestBody:
+        required: true
+        content:
+          application/fhir+json:
+            schema:
+              $ref: 'components/schemas/eRS-CancelReferral-Request-Parameters.yaml'
+            examples:
+              basic-example:
+                summary: Cancel Referral with Reason and Comment
+                value:
+                  $ref: 'components/examples/cancelReferral/requests/cancelReferralExample.json'
+      responses:
+        '200':
+          $ref: 'components/schemas/responses/cancelReferral/200Response.yaml'
+        '400':
+          $ref: 'components/schemas/responses/BadRequest.yaml'
```

Comment:
Would this endpoint return a 400 response as it accepts a request body?

Reply:
Yes it returns 400 if general FHIR validation fails 

i.e. ERSSUP-57567 - 1.4. eRS performs initial generic validation of the FHIR message received and finds this to be satisfactory via [ERSSUP-11958](https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-11958) (failure conditions covered in [ERSSUP-11958](https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-11958))

In ERSSUP-11958, Failure - Malformed JSON returns 400.